zip-0227.rst

```diff
@@ -1,8 +1,359 @@
+- Furthermore, NFT issuance is enabled by issuing in a single bundle several issuance actions, where each `AssetId` corresponds to `value = 1` at the fundamental unit level. Issuers and users should make sure that `finalize = 1` for each of the actions in this scenario.
+
+Consensus Rule Changes
+----------------------
+
+For the IssueBundle,
+- Verify the RedPallas-based issuance authorization signature on `SIGHASH`, `authorization` is valid, based on `authorization.VerifySig(ik, SIGHASH)`
+
+For each `IssueAction` in `IssueBundle`:
+- check that `asset_desc` is a string with `0 < byte size <= 512`
+- retrieve `AssetId` from the first note in the sequence and check that `AssetId = GroupHash^{\mathbb{P}}(ik || asset_desc)` is properly derived
+- check that the `AssetId` does not exist in the `previously_finalized` set in the global state
+- check that every note in the `IssueAction` contains the same `AssetId` and also that every note is of type `Note` and is properly constructed as :math:`note = (\mathsf{g_d, pk_d, v, \rho, \psi, AssetId})`
+
+If all of the above checks pass, do the following:
+- For each note, compute the note commitment as :math:`cm = \mathsf{NoteCommit^{OrchardZSA}_{rcm}(repr_{\mathbb{P}}(g_d), repr_{\mathbb{P}}(pk_d), v, \rho, \psi, AssetId)}` and
+- add `cm` to the Merkle tree of note commitments, `NoteCommitmentTree`
+- If `finalize = 1`, add `AssetId` to the `previously_finalized` set in the global state
+
+Rationale
+=========
+The following is a list of rationale for different decisions made in the proposal:
+
+- The issuance key structure was kept aligned to the original key tree in order to ease issuance integration with wallets and to prevent users from having to manage multiple keys and secrets.
+- the `asset_desc` is a general byte string in order to allow for a wide range of information type to be included that may be associated with the assets. Some are:
+    - links for storage such as for NFTs
+    - other metadata for native assets
+    - bridging information for wrapped assets (chain of origin, issuer name, etc)
+    - information to be committed by the issuer, though not enforceable by the protocol
+
+TxId Digest
+===========
+A new issuance transaction digest algorithm is defined that constructs the identifier for an issuance transaction. Each branch of the tree of hashes will correspond to a specific subset of transaction data. The overall structure of the hash is as follows; each name referenced here will be described in detail below::
+
+    issuance_txid_digest
+    ├── issueActions
+    └── issuerVerificationKey
+
+In the specification below, nodes of the tree are presented in depth-first order.
+
+issuance_txid_digest
+--------------------
+A BLAKE2b-256 hash of the following values ::
+
+   T.1: issueActions            (field encoding bytes)
+   T.2: issuerVerificationKey   (field encoding bytes)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdOrcZSAIssue"
```

Comment:
The SIGHASH changes should become a new section in ZIP-244, with a note that it will be deployed in whichever network upgrade includes ZSAs:
https://zips.z.cash/zip-0244

Reply:
As per our syncs, we will have sections in our ZIPs (226 and 227) detailing the changes to `txid_digest` and `SIGHASH`, and the changes to ZIP 244 (or migration to a newer ZIP) will be done subsequently